exam/models/exam.py

```diff
@@ -74,57 +82,148 @@ class ExamExam(models.Model):
 
     @api.constrains('start_date', 'end_date')
     def check_date_exam(self):
-        if self.end_date < self.start_date:
-            raise UserError(_('End date of Exam should be \
-                              greater than start date'))
-
+        '''Method to check constraint of exam start date and end date'''
+        for rec in self:
+            if rec.end_date < rec.start_date:
+                raise ValidationError(_('Exam end date should be \
+                                  greater than start date'))
+            for line in rec.exam_schedule_ids:
+                if line.timetable_id:
+                    for tt in line.timetable_id.exam_timetable_line_ids:
+                        if not rec.start_date <= tt.exm_date <= rec.end_date:
```

Comment:
It is working well,
it  check the exam_date is not in between the start date and end date of the exam schedule.